components/eventing-controller/pkg/handlers/sink/validator.go

```diff
@@ -0,0 +1,99 @@
+package sink
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	"github.com/kyma-project/kyma/components/eventing-controller/controllers/events"
+	"github.com/kyma-project/kyma/components/eventing-controller/logger"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const clusterLocalURLSuffix = "svc.cluster.local"
+
+type Validator interface {
+	Validate(subscription *v1alpha1.Subscription) error
+}
+
+type defaultSinkValidator struct {
+	ctx      context.Context
+	client   client.Client
+	recorder record.EventRecorder
+	logger   *logger.Logger
+}
+
+// compile-time check
+var _ Validator = &defaultSinkValidator{}
+
+func NewSinkValidator(ctx context.Context, client client.Client, recorder record.EventRecorder, logger *logger.Logger) Validator {
```

Comment:
`sink.NewSinkValidator` could be renamed to `sink.NewValidator`. The package name is already mentioning the sink